src/v8/inspector/mod.rs

```diff
@@ -0,0 +1,425 @@
+//! callback must be properly set.
+//!
+//! # Sessions
+//!
+//! To be able to debug remotely, a server is required. Here one may
+//! find the [server::WebSocketServer] useful. The
+//! [server::WebSocketServer] is a very simple WebSocket server, which
+//! can be used with an [Inspector].
+//!
+//! To start a debugging server session, it is required to have a
+//! [V8ContextScope], a [server::WebSocketServer] and an [Inspector].
+//! All of those are gently packed within an easy-to-use
+//! [server::DebuggerSession] struct which abstracts all the work. It is
+//! a scoped object, meaning that once an object of
+//! [server::DebuggerSession] leaves its scope, the debugger server and
+//! the debugging session are stopped.
+//!
+//! In case the `"debug-server"` feature isn't enabled, the user of the
+//! crate must manually provide a way to receive and send messages over
+//! the network and feed the [Inspector] with data.
+use std::{ops::Deref, ptr::NonNull, sync::Arc};
+
+#[cfg(feature = "debug-server")]
+pub mod messages;
+#[cfg(feature = "debug-server")]
+pub mod server;
+
+use crate::v8_c_raw::bindings::v8_context_ref;
+
+use super::{isolate::V8Isolate, v8_context_scope::V8ContextScope};
+
+/// The debugging inspector, carefully wrapping the
+/// [`v8_inspector::Inspector`](https://chromium.googlesource.com/v8/v8/+/refs/heads/main/src/inspector)
+/// API. An inspector is tied to the [V8Isolate] it was created for.
+///
+/// # Example
+///
+/// ```rust
+/// use v8_rs::v8::*;
+/// use v8_rs::v8::inspector::RawInspector;
+///
+/// // Initialise the V8 engine:
+/// v8_init_platform(1, Some("--expose-gc")).unwrap();
+/// v8_init().unwrap();
+///
+/// // Create a new isolate:
+/// let isolate = isolate::V8Isolate::new();
+///
+/// // Enter the isolate created:
+/// let i_scope = isolate.enter();
+///
+/// // Create a JS execution context for code invocation:""
+/// let ctx = i_scope.new_context(None);
+///
+/// // Enter the created execution context for debugging:
+/// let ctx_scope = ctx.enter(&i_scope);
+///
+/// // Create an inspector.
+/// let _inspector = RawInspector::new(&ctx_scope);
+/// ```
+#[derive(Debug)]
+pub struct RawInspector {
+    raw: NonNull<crate::v8_c_raw::bindings::v8_inspector_c_wrapper>,
+}
+
+impl RawInspector {
+    /// Creates a new inspector for the provided isolate.
+    #[allow(clippy::not_unsafe_ptr_arg_deref)]
+    pub fn new(context_scope: &V8ContextScope<'_, '_>) -> Self {
+        let raw_context = context_scope.get_inner();
+        let raw = unsafe {
+            NonNull::new_unchecked(crate::v8_c_raw::bindings::v8_InspectorCreate(
+                raw_context,
+                None,
+                std::ptr::null_mut(),
+                None,
+                std::ptr::null_mut(),
+            ))
+        };
+        Self { raw }
+    }
+
+    /// Returns the isolate this inspector is bound to. The isolate
+    /// returned won't be released automatically.
+    pub fn get_isolate(&self) -> V8Isolate {
```

Comment:
Why do we need this API? I believe this API is not safe, user can hold the isolate even after the isolate has been freed.